src/v/cluster/errc.h

```diff
@@ -62,7 +62,6 @@ enum class errc : int16_t {
     invalid_request,
     no_update_in_progress,
     unknown_update_interruption_error,
-    invalid_retention_configuration,
```

Comment:
Is removing this a compatibility concern?

Reply:
Not as far as I can see: we don't expose this specific error code externally beyond its string-ized form.

Reply:
I suppose, I guess at worst this results in seeing the wrong string if we get this error in a mixed-version cluster.

Reply:
Taking a step back, another question is also how this affects compat for
```
    throttling_quota_exceeded,
    cluster_already_exists,
```
which are now one value lower

Reply:
Fair point.  I _think_ these codes are super unlikely to come up while we're in the mixed version state... with a bit more time I would honestly err on the side of caution and leave the old code in to avoid even having to think about it, but I think we're optimizing for speed right now.